### PR TITLE
Exchange rate implementation

### PR DESCRIPTION
**Pull request by Milan-Cerovsky**
# Author

- [x] **Self-review** your own code in GitHub's web interface[^1]
- [ ] Add **automated tests** as appropriate
- [ ] Update the [**manual tests**](../blob/main/docs/testing/manual_testing)[^2] as appropriate
- [ ] Check the **code coverage**[^3] report for the automated tests
- [x] Update **documentation** as appropriate (e.g [**README.md**](../blob/main/README.md), [**Architecture.md**](../blob/main/docs/Architecture.md), [**CHANGELOG.md**](../blob/main/CHANGELOG.md), etc.)
- [x] **Run the app** and try the changes
- [ ] Pull in the latest changes from the **main** branch and **squash** your commits before assigning a reviewer[^4]

# Reviewer

- [x] Check the code with the [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md) **checklist**
- [ ] Perform an **ad hoc review**[^5]
- [x] Review the **automated tests**
- [x] Review the **manual tests**
- [x] Review the **documentation**, [**README.md**](../blob/main/README.md), [**Architecture.md**](../blob/main/docs/Architecture.md), etc. as appropriate
- [x] **Run the app** and try the changes[^6]

[^1]: _Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs._
[^2]: _While we aim for automated testing of the application, some aspects require manual testing. If you had to manually test something during development of this pull request, write those steps down._
[^3]: _While we are not looking for perfect coverage, the tool can point out potential cases that have been missed. Code coverage can be generated with: `./gradlew check` for Kotlin modules and `./gradlew connectedCheck -PIS_ANDROID_INSTRUMENTATION_TEST_COVERAGE_ENABLED=true` for Android modules._
[^4]: _Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit._
[^5]: _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
[^6]: _While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data. Perform this step last, after verifying the code changes are safe to run locally._

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -131,7 +131,7 @@ ZCASH_EMULATOR_WTF_API_KEY=
 # Optional absolute path to a Zcash SDK checkout.
 # When blank, it pulls the SDK from Maven.
 # When set, it uses the path for a Gradle included build.  Path can either be absolute or relative to the root of this app's Gradle project.
-SDK_INCLUDED_BUILD_PATH=
+SDK_INCLUDED_BUILD_PATH=/Users/milancerovsky/Developer/zcash/zcash-android-wallet-sdk
```
Comment (HonzaR) [inline gradle.properties:134]: Mistakenly added

Comment (Milan-Cerovsky) [reply gradle.properties:134]: will remove once SDK is done

File: ui-design-lib/src/main/java/co/electriccoin/zcash/ui/design/theme/ExchangeRateColors.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
+@file:Suppress("MagicNumber")
+
+package co.electriccoin.zcash.ui.design.theme
+
+import androidx.compose.runtime.Immutable
+import androidx.compose.runtime.staticCompositionLocalOf
+import androidx.compose.ui.graphics.Color
+
+@Immutable
+data class ExchangeRateColors(
```
Comment (HonzaR) [inline ui-design-lib/src/main/java/co/electriccoin/zcash/ui/design/theme/ExchangeRateColors.kt:10]: As discussed internally, we should split this class make it part of the new alternative `ZashiColors` and make it internal

File: ui-design-lib/src/main/java/co/electriccoin/zcash/ui/design/theme/ZashiColors.kt [hunk 3]
```diff
@@ -0,0 +1,67 @@
+@file:Suppress("MagicNumber")
+
+package co.electriccoin.zcash.ui.design.theme
+
+import androidx.compose.runtime.Immutable
+import androidx.compose.runtime.staticCompositionLocalOf
+import androidx.compose.ui.graphics.Color
+
+@Immutable
+data class ZashiColors(
```
Comment (HonzaR) [inline ui-design-lib/src/main/java/co/electriccoin/zcash/ui/design/theme/ZashiColors.kt:10]: Let's file and link here a follow-up on extracting the color values to dedicated values so we can reuse them as we do in the original `Color.kt` approach

File: ui-design-lib/src/main/java/co/electriccoin/zcash/ui/design/theme/ZcashTheme.kt [hunk 4]
```diff
@@ -27,22 +27,16 @@ fun ZcashTheme(
     content: @Composable () -> Unit
 ) {
     val useDarkMode = forceDarkMode || isSystemInDarkTheme()
+    val baseColors = if (useDarkMode) DarkColorPalette else LightColorPalette
+    val extendedColors = if (useDarkMode) DarkExtendedColorPalette else LightExtendedColorPalette
+    val zashiColors = if (useDarkMode) DarkZashiColorPalette else LightZashiColorPalette
+    val exchangeRateColors = if (useDarkMode) DarkExchangeRateColorPalette else LightExchangeRateColorPalette
```
Comment (HonzaR) [inline ui-design-lib/src/main/java/co/electriccoin/zcash/ui/design/theme/ZcashTheme.kt:33]: If we follow the comments above, we can then remove `exchangeRateColors` here.

File: ui-lib/build.gradle.kts [hunk 5]
```diff
@@ -120,6 +122,7 @@ dependencies {
     implementation(projects.sdkExtLib)
     implementation(projects.spackleAndroidLib)
     api(projects.uiDesignLib)
+    implementation("androidx.graphics:graphics-shapes-android:1.0.0-rc01")
```
Comment (HonzaR) [inline ui-lib/build.gradle.kts:125]: Is this new dependency necessary? If so, please move it into the `gradle.properties` and `settings.gradle` files.

**Review by HonzaR — COMMENTED** [review]
Flushing the first few review comments. Thanks for taking a look into them.

File: ui-design-lib/src/main/java/co/electriccoin/zcash/ui/design/theme/ZashiColors.kt (hunk 3 above)
Comment (Milan-Cerovsky) [reply ui-design-lib/src/main/java/co/electriccoin/zcash/ui/design/theme/ZashiColors.kt:10]: done

File: ui-design-lib/src/main/java/co/electriccoin/zcash/ui/design/theme/ZcashTheme.kt (hunk 4 above)
Comment (Milan-Cerovsky) [reply ui-design-lib/src/main/java/co/electriccoin/zcash/ui/design/theme/ZcashTheme.kt:33]: done

File: ui-lib/build.gradle.kts (hunk 5 above)
Comment (Milan-Cerovsky) [reply ui-lib/build.gradle.kts:125]: it can be removed :)

File: ui-lib/src/main/java/co/electriccoin/zcash/ui/MainActivity.kt [hunk 6]
```diff
@@ -150,6 +159,54 @@ class MainActivity : FragmentActivity() {
         }
     }
 
+    private fun setContent(
```
Comment (HonzaR) [inline ui-lib/src/main/java/co/electriccoin/zcash/ui/MainActivity.kt:162]: We'll extract this new logic into a separate PR and provide documentation for it.

Comment (Milan-Cerovsky) [reply ui-lib/src/main/java/co/electriccoin/zcash/ui/MainActivity.kt]: done

File: ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/balances/model/WalletDisplayValues.kt [hunk 7]
```diff
@@ -35,9 +35,6 @@ data class WalletDisplayValues(
             var statusText = ""
             var statusAction: StatusAction = StatusAction.None
 
-            // TODO [#578]: Provide Zatoshi -> USD fiat currency formatting
```
Comment (HonzaR) [inline ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/balances/model/WalletDisplayValues.kt:38]: Don't forget to close this SDK issue once the PR is merged.

File: ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/exchangerate/BaseExchangeRateOptIn.kt
Comment (HonzaR) [inline ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/exchangerate/BaseExchangeRateOptIn.kt:1]: We should ideally use predefined sizes from `Dimens` as much as possible

File: ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/exchangerate/SecondaryCard.kt [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+import co.electriccoin.zcash.ui.design.theme.ZcashTheme
+
+@Composable
+internal fun SecondaryCard(
+    modifier: Modifier = Modifier,
+    content: @Composable ColumnScope.() -> Unit,
+) {
+    Card(
+        modifier = modifier,
+        shape = RoundedCornerShape(12.dp),
```
Comment (HonzaR) [inline ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/exchangerate/SecondaryCard.kt:20]: Again, let's use or define size for this in `Dimens`.

File: ui-lib/src/main/java/co/electriccoin/zcash/ui/common/compose/BalanceWidget.kt
[the remark this answers is not in the document]
Comment (Milan-Cerovsky) [reply ui-lib/src/main/java/co/electriccoin/zcash/ui/common/compose/BalanceWidget.kt:1]: done

**Comment by Milan-Cerovsky** [comment]
Dimensions will be done in #1555

**Review by HonzaR — APPROVED** [review]
utACK. Left a few more comments inline.
